packages/apollo-server-core/src/ApolloServer.ts

```diff
@@ -413,6 +409,11 @@ export class ApolloServerBase {
     } else {
       throw new Error("Unexpected error: Unable to resolve a valid GraphQLSchema.  Please file an issue with a reproduction of this error, if possible.");
     }
+
+    // Plugins will be instantiated if they aren't already, and this.plugins
+    // is populated accordingly.
+    this.ensurePluginInstantiation(plugins);
```

Comment:
Correct.  It's been a minute now, but I believe the specific reason was because of the way we negotiate the schema in a gateway (eventually resolved after we fetch and/or compose) vs non-gateway mode (immediately, because it's passed in or we construct it from `typeDefs` SDL).

Seemed like a good thing for it to have access to, generally, but somehow proved to be a necessity to make our existing plugins work with the new patterns.  (Again, it's been a minute now, sorry for the lack of precision as to what that configuraion was now.)